internal/common/conversion/string_framework.go

```diff
@@ -24,6 +24,15 @@ func ToTFMapOfSlices(ctx context.Context, values map[string][]string) (basetypes
 	return types.MapValueFrom(ctx, types.ListType{ElemType: types.StringType}, values)
 }
 
+func ToTFMapOfString(ctx context.Context, diags *diag.Diagnostics, values *map[string]string) basetypes.MapValue { //nolint:gocritic // easy usage from sdk fields
```

Comment:
!extra conversion function to simplify model_xxx.!

Reply:
what's the linter issue? can't it be avoided?

Reply:
ptrToRefParam: consider `values` to be of non-pointer type (gocritic)

Reply:
It is hard to avoid. Consider:
```golang
		AwsPrivateLink:    conversion.ToTFMapOfString(ctx, diags, input.AwsPrivateLink),
		AwsPrivateLinkSrv: conversion.ToTFMapOfString(ctx, diags, input.AwsPrivateLinkSrv),
```
The `input.AwsPrivateLink` is a pointer. The alternative of using values as a normal struct requires `nil` checks in all caller sights.

Also consider:
```golang
ContainerId:   conversion.ToTFMapOfString(ctx, diags, &todoContainerID),
```
The ease of using `&` for normal struct variables

Reply:
can we just use `values map[string]string`  without pointer?    map zero value is nil, which is different from an empty map, so no real need to pass a pointer here, we can differentiate map directly.

![value](https://github.com/user-attachments/assets/0464d39f-7958-4a4b-9767-3ab95474aa52)

difference:
![nil](https://github.com/user-attachments/assets/f96e0f6c-992e-40e5-a311-156a48e51927)

Reply:
> The input.AwsPrivateLink is a pointer. The alternative of using values as a normal struct requires nil checks in all caller locations.

similarly, do we need pointers in the caller or a map is enough?

Reply:
Isn't it unsafe to do `*input.AwsPrivateLink`?

Reply:
yes, it's unsafe, my follow-up question is whether we need a pointer there as well

Reply:
I'm not sure I understand. The pointer comes from the SDK

Reply:
Resolved offline